drv/ArrayList.drv

```diff
@@ -587,6 +587,40 @@ public class ARRAY_LIST KEY_GENERIC extends ABSTRACT_LIST KEY_GENERIC implements
 		return modified;
 	}
 
+#ifdef JDK_PRIMITIVE_PREDICATE	
+	/** Remove from this collection all elements which satisfy the given predicate.
+	 *
+	 * @param filter a predicate which returns {@code true} for elements to be
+	 *        removed.
+	 * @see Collection#removeIf(java.util.function.Predicate)
+	 * @return {@code true} if any elements were removed.
+	 */
+	public boolean removeIf(final JDK_PRIMITIVE_PREDICATE filter) {
+		java.util.Objects.requireNonNull(filter);
+		final KEY_TYPE[] a = this.a;
+		int j = 0;
+		for(int i = 0; i < size; i++)
+			if (!filter.test(a[i])) a[j++] = a[i];
+		final boolean modified = size != j;
+		size = j;
+		return modified;
+	}
+#endif
+
+#if KEYS_REFERENCE
+	public boolean removeIf(final java.util.function.Predicate<? super KEY_GENERIC_CLASS> filter) {
+		java.util.Objects.requireNonNull(filter);
+		final KEY_GENERIC_TYPE[] a = this.a;
+		int j = 0;
+		for(int i = 0; i < size; i++)
+			if (!filter.test(a[i])) a[j++] = a[i];
+		Arrays.fill(a, j, size, null);
```

Comment:
This can be omitted I guess. Everything in a after `size` will not be read.

Reply:
No, the clear function even clears the array too. Otherwise objects won't be de-allocated by java if there is a reference left.
for primitives yes you can leave them, but this is the Object version only.